core/src/main/scala/kafka/server/BrokerToControllerChannelManager.scala

```diff
@@ -159,12 +161,12 @@ class BrokerToControllerChannelManagerImpl(
   config: KafkaConfig,
   channelName: String,
   threadNamePrefix: Option[String],
-  retryTimeoutMs: Long
+  retryTimeoutMs: Long,
+  currentNodeControllerApiVersions: Option[NodeApiVersions]
 ) extends BrokerToControllerChannelManager with Logging {
   private val logContext = new LogContext(s"[BrokerToControllerChannelManager broker=${config.brokerId} name=$channelName] ")
   private val manualMetadataUpdater = new ManualMetadataUpdater()
   private val apiVersions = new ApiVersions()
-  private val currentNodeApiVersions = NodeApiVersions.create()
```

Comment:
This is the source of the bug, we always assume currentNodeApiVersions=`ApiKeys.zkBrokerApis`